easybuild/toolchains/compiler/gcc.py

```diff
@@ -110,3 +117,43 @@ def _set_compiler_vars(self):
                 raise EasyBuildError("Failed to determine software root for GCC")
 
         self.variables.append_subdirs("LDFLAGS", gcc_root, subdirs=["lib64", "lib"])
+
+    def _set_optimal_architecture(self, default_optarch=None):
+        """
+        GCC-specific adjustments for optimal architecture flags.
+
+        :param default_optarch: default value to use for optarch, rather than using default value based on architecture
+                                (--optarch and --optarch=GENERIC still override this value)
+        """
+        if default_optarch is None and self.arch == systemtools.AARCH64:
+            # On AArch64, -mcpu=native is not supported prior to GCC 6.  In this case, try to optimize for the detected
+            # CPU model (vanilla ARM cores only).  Note that this heuristic may fail if the CPU model is not supported
+            # by the GCC version being used.
+            gcc_version = self.get_software_version(self.COMPILER_MODULE_NAME)[0]
```

Comment:
Ah, I see why this worked: we usually make sure in the `GCC` bundles that `$EBROOTGCC` and `$EBVERSIONGCC` are also defined for the `GCCcore` module, by defining `altroot`/`altversion` in the `GCC` bundle easyconfig, see https://github.com/hpcugent/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/g/GCC/GCC-6.2.0-2.27.eb#L21

However, it would be better to take both `GCCcore` (first) and `GCC` (second)` into account for determining the GCC version, like we already do for the installation prefix in `_set_compiler_vars`.

Are you up for making that change @geimer?